### PR TITLE
Fix tony not using boss effects.

### PR DESCRIPTION
**Pull request by TokyoSU**
- Fixed some comment of @Lwmte from PR #979 
- Require change from GetRandomControl() to Random::GenerateInt() @Sezzary
- Added some description and renamed TonyFlameType enum values.
- Removed GetFlagField and SetFlagField from Tony code, they will be removed later to avoid confusion.
- Added boss effects which tony was lacking.

### CODE REVIEW

File: TombEngine/Objects/TR3/Entity/TonyBoss.cpp [hunk 1]
```diff
@@ -25,55 +25,70 @@ namespace TEN::Entities::Creatures::TR3
...
 	{
+		TONY_ANIM_WAIT, // Wait lara.
+		TONY_ANIM_RISE,
+		TONY_ANIM_FLY, // Real idle state.
+		TONY_ANIM_SHOOT_FIRE_RIGHT_HAND,
+		TONY_ANIM_SHOOT_CEILING, // with two hand.
+		TONY_ANIM_FLIPMAP, // Causes an shockwave explosion.
+		TONY_ANIM_DEATH
+	};
+
```
Comment (Nemoel-Tomo) [inline TombEngine/Objects/TR3/Entity/TonyBoss.cpp:70]: 70 and 71 double blank line

Comment (TokyoSU) [reply TombEngine/Objects/TR3/Entity/TonyBoss.cpp:73]: Fixed by @Sezzary :)

**Comment by Adngel** [comment]
I had not any issue with Tony effects using this branch.

However one of the Tony attacks keep failing to hit Lara. (There straight fireball, is shoot at very high height so even if Lara is at same floor height than Tony, the firewall flies over here without touch Lara).


Aditionally, would be possible that Tony used the fire flame sprites for his fire effects? (Don't know very well how that work so I don't know if it's too difficult change or not).

**Comment by TokyoSU** [comment]
> I had not any issue with Tony effects using this branch.
> 
> However one of the Tony attacks keep failing to hit Lara. (The straight fireball, is shoot at very high height so even if Lara is at same floor height than Tony, the firewall flies over her without touching Lara).

Yes i have this issue too. i will fix it :)

**Comment by Kubsy** [comment]
I see the whole tony cpp file delete and replaced with a new. Tony still works right?

**Review by Kubsy — APPROVED** [review]

**Comment by TokyoSU** [comment]
> I see the whole tony cpp file delete and replaced with a new. Tony still works right?

Yes, the new name is the coding convention which is applied.

**Review by Adngel — APPROVED** [review]

**Pull request merged**